library/core/src/ffi/c_str.rs

```diff
@@ -449,38 +445,59 @@ impl CStr {
     /// behavior when `ptr` is used inside the `unsafe` block:
     ///
     /// ```no_run
-    /// # #![allow(unused_must_use)]
-    /// # #![expect(dangling_pointers_from_temporaries)]
-    /// use std::ffi::CString;
-    ///
     /// // Do not do this:
-    /// let ptr = CString::new("Hello").expect("CString::new failed").as_ptr();
```

Comment:
Here, we called `as_ptr()` on CString, but it autoderefs to CStr when looking up where to call. The current example does in fact illustrate usage of `CStr::as_ptr`, and the issue was mistaken. The review from https://github.com/rust-lang/rust/pull/84000 seems to apply in full to this example.